test/functional/options/defaults_spec.lua

```diff
@@ -914,3 +929,14 @@ describe('stdpath()', function()
     end)
   end)
 end)
+
+describe('tempname()', function()
+  describe('returns a string', function()
```

Comment:
don't need a new describe() for this, it should be sibling to the other test above

Reply:
I'm not sure how to word the check but I've moved it into place

Reply:
Took me way too long to realise I could use `PATHSEPSTR` to make this just work.

I also added a check for `APPNAME == "."` I feel that would break things